packages/app/src/systems/DApp/machines/addNetworkRequestMachine.tsx

```diff
@@ -0,0 +1,148 @@
+  };
+};
+
+type MachineEvents =
+  | {
+      type: 'START';
+      input: AddNetworkInputs['start'];
+    }
+  | { type: 'APPROVE'; input: void }
+  | { type: 'REJECT'; input: void };
+
+export const addNetworkRequestMachine = createMachine(
+  {
+    predictableActionArguments: true,
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import('./addNetworkRequestMachine.typegen').Typegen0,
+    schema: {
+      context: {} as MachineContext,
+      services: {} as MachineServices,
+      events: {} as MachineEvents,
+    },
+    id: '(machine)',
+    initial: 'idle',
+    states: {
+      idle: {
+        on: {
+          START: {
+            actions: ['assignStartData'],
+            target: 'reviewNetwork',
+          },
+        },
+      },
+      reviewNetwork: {
+        on: {
+          APPROVE: {
+            target: 'addingNetwork',
+          },
+          REJECT: {
+            actions: [assignErrorMessage('Rejected request!')],
+            target: 'failed',
+          },
+        },
+      },
+      addingNetwork: {
+        invoke: {
+          src: 'saveNetwork',
+          data: {
+            input: (ctx: MachineContext) => ({
+              data: ctx.network,
+            }),
+          },
+          onDone: [
+            FetchMachine.errorState('failed'),
+            {
+              actions: ['notifyRefreshNetworks', 'notifyUpdateAccounts'],
+              target: 'done',
+            },
+          ],
+        },
+      },
+      done: {
+        type: 'final',
+      },
+      failed: {},
+    },
+  },
+  {
+    actions: {
+      assignStartData: assign((ctx, ev) => ({
+        ...ctx,
+        origin: ev.input.origin,
+        title: ev.input.title,
+        favIconUrl: ev.input.favIconUrl,
+        network: ev.input.network,
+      })),
+      notifyRefreshNetworks: () => {
+        store.refreshNetworks();
+      },
+      notifyUpdateAccounts: () => {
+        store.updateAccounts();
+      },
+    },
+    services: {
+      saveNetwork: FetchMachine.create<
+        NetworkInputs['addNetwork'],
+        MachineServices['saveNetwork']['data']
+      >({
+        showError: true,
+        async fetch({ input }) {
+          try {
+            if (!input?.data) {
+              throw new Error('Invalid network');
+            }
+            const createdNetwork = await NetworkService.addNetwork(input);
+            if (!createdNetwork) {
+              throw new Error('Failed to add network');
+            }
+            const selectedNetwork = await NetworkService.selectNetwork({
+              id: createdNetwork.id!,
+            });
+            return selectedNetwork;
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } catch (error: any) {
+            if (error?.message.includes('uniqueness')) {
+              throw new Error('This network Name or URL already exists');
+            }
+            throw error;
+          }
+        },
```

Comment:
clean it as now `NetworkService.addNetwork` will do the validations using `NetworkService.assertAddNetwork`